src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -129,20 +133,32 @@ export class ProductCard extends PureComponent {
         );
     }
 
+    renderVisualOption({ value, label, type }, i) {
+        const _type = type === OPTION_TYPE_COLOR;
```

Comment:
Rename this variable, the current name does not reflect the variable's purpose. Also, make sure that Color and String options indeed are the only two types of options you need to handle here.